src/Microsoft.AspNetCore.SignalR.Protocols.Json/Protocol/JsonHubProtocol.cs

```diff
@@ -599,14 +605,48 @@ private HubMessage BindInvocationMessage(string invocationId, string target, obj
             return new InvocationMessage(invocationId, target, arguments);
         }
 
+        private static bool ReadForType(JsonTextReader reader, Type type)
+        {
+            // Explicity read values as dates from JSON with reader.
+            // We do this because otherwise dates are read as strings
+            // and the JsonSerializer will use a conversion method that won't
+            // preserve UTC in DateTime.Kind for UTC ISO8601 dates
+            if (type == typeof(DateTime) || type == typeof(DateTime?))
+            {
+                reader.ReadAsDateTime();
```

Comment:
This changes the token type to Date right? Do we have anything else like this?

Reply:
Yes.

What?

Reply:
I was looking at `JsonTokenType` and JsonSerializerSettings to see if there are any other things in this category. The only one I see is FloatParseHandling.

Reply:
Only date reading was disable in the reader.